test/run_pass/test_uart.js

```diff
@@ -35,18 +36,21 @@ var serial = new uart(device, options, function(err){
     read = 1;
 
     if (read && write) {
-      serial.close();
+      serial.closeSync();
+      console.log('close done');
     }
   });
 
+  serial.writeSync("Hello IoT.js.\n\r");
 
-  serial.write("Hello there?\n", function(err) {
+  serial.write("Hello there?\n\r", function(err) {
```

Comment:
It has nothing to do with osx. Characters can not be printed in the first column without `\r` in serial.